File: refactored/duplicate-assert/CSharpXUnit/CSharpXUnit.Tests/Tests/TestUUIDService_testChildId.cs
public class TestUUIDService
{
    [Fact]
    public void TestChildId1()
    {
        SetSystemProperty(UUIDService.CONF_GENERATOR, "counter");
        var services = new Services();
        services.Init();
        try
        {
            var uuid = services.Get<UUIDService>();
            string id = uuid.GenerateId(ApplicationType.WORKFLOW);
            string childId = uuid.GenerateChildId(id, "a");
            Assert.Equal(id, uuid.GetId(childId));
            Assert.Equal("a", uuid.GetChildName(childId));
        }
        finally
        {
            services.Destroy();
        }
    }

    [Fact]
    public void TestChildId2()
    {
        SetSystemProperty(UUIDService.CONF_GENERATOR, "counter");
        var services = new Services();
        services.Init();
        try
        {
            var uuid = services.Get<UUIDService>();
            string id = uuid.GenerateId(ApplicationType.WORKFLOW);
            string childId = uuid.GenerateChildId(id, "a");
        }
        finally
        {
            services.Destroy();
        }

        SetSystemProperty(UUIDService.CONF_GENERATOR, "random");
        services = new Services();
        services.Init();
        try
        {
            var uuid = services.Get<UUIDService>();
            string id = uuid.GenerateId(ApplicationType.WORKFLOW);
            string childId = uuid.GenerateChildId(id, "a");
            Assert.Equal(id, uuid.GetId(childId));
            Assert.Equal("a", uuid.GetChildName(childId));
        }
        finally
        {
            services.Destroy();
        }
    }
}

File: refactored/duplicate-assert/python-pytest/TestUUIDService_testChildId.py
import pytest

def test_child_id_1():
    set_system_property(UUIDService.CONF_GENERATOR, "counter")
    services = Services()
    services.init()
    try:
        uuid = services.get(UUIDService)
        id = uuid.generate_id(ApplicationType.WORKFLOW)
        child_id = uuid.generate_child_id(id, "a")
        assert id == uuid.get_id(child_id)
        assert "a" == uuid.get_child_name(child_id)
    finally:
        services.destroy()

def test_child_id_2():
    set_system_property(UUIDService.CONF_GENERATOR, "counter")
    services = Services()
    services.init()
    try:
        uuid = services.get(UUIDService)
        id = uuid.generate_id(ApplicationType.WORKFLOW)
        child_id = uuid.generate_child_id(id, "a")
    finally:
        services.destroy()

    set_system_property(UUIDService.CONF_GENERATOR, "random")
    services = Services()
    services.init()
    try:
        uuid = services.get(UUIDService)
        id = uuid.generate_id(ApplicationType.WORKFLOW)
        child_id = uuid.generate_child_id(id, "a")
        assert id == uuid.get_id(child_id)
        assert "a" == uuid.get_child_name(child_id)
    finally:
        services.destroy()
